### PR TITLE
FIX: S3C-1087 put and copy metadata for external backends

### PR DESCRIPTION
**Pull request by nicolas2bert**
- When copying to Azure we want to keep metadata up to date on the Azure side.

- Put and copying to Azure we want to trim `x-amz-meta`prefix from metadata header keys.

### CODE REVIEW

File: lib/data/external/utils.js [hunk 1]
```diff
@@ -112,6 +112,25 @@ const utils = {
         }
         return cb();
     },
+    translateMetaHeaders(metaHeaders, tags) {
```
Comment (electrachong) [inline lib/data/external/utils.js]: this only works for azure right? maybe translateAzureMetaHeaders?

**Comment by nicolas2bert** [comment]
@ironman-machine r+

**Comment by nicolas2bert** [comment]
@ironman-machine r+

**Comment by nicolas2bert** [comment]
@ironman-machine r+

**Pull request merged**